File: mahasiswa.py
def nilai_huruf(nilaiAkhir):
    if 90 <= nilaiAkhir <= 100:
        return 'A'
    elif 70 <= nilaiAkhir < 90:
        return 'B'
    elif 60 <= nilaiAkhir < 70:
        return 'C'
    elif 50 <= nilaiAkhir < 60:
        return 'D'
    return 'E'
nim = input("Masukkan NIM: ")
nama = input("Masukkan Nama: ")
nilai_uts = float(input("masukkan nilai UTS: "))
nilai_uas = float(input("masukkan nilai UAS: "))
nilai_tugas = float(input("masukkan nilai tugas: "))
nilai_akhir = (0.3 * nilai_uts) + (0.5 * nilai_uas) + (0.2 * nilai_tugas)
print("\nLAPORAN NILAI MAHASiSWA")
print(f"NIM: {nim}")
print(f"nama: {nama}")
print(f"nilai Akhir : {nilai_akhir}")
print(f"Nilai Huruf : {nilai_huruf(nilai_akhir)}")

File: mahasiswa.c
#include <stdio.h>

char nilaiHuruf(float nilaiAkhir) {
    if (nilaiAkhir >= 90 && nilaiAkhir <= 100)
        return 'A';
    else if (nilaiAkhir >= 70 && nilaiAkhir < 90)
        return 'B';
    else if (nilaiAkhir >= 60 && nilaiAkhir < 70)
        return 'C';
    else if (nilaiAkhir >= 50 && nilaiAkhir < 60)
        return 'D';
  
        return 'E'; // Invalid
}
        int main() {
            char nim[20], nama[50];
            float nilaiUTS, nilaiUAS, nilaiTugas, nilaiAkhir;
            printf("masukkan NIM : ");
            scanf("%s", nim);
            printf("masukkan nama : ");
            scanf(" %[^\n]s", nama);
            printf("masukkan nilai UTS : ");
            scanf("%f", &nilaiUTS);
            printf("masukkan nilai UAS : ");
            scanf("%f", &nilaiUAS);
            printf("masukkan nilai Tugas : ");
            scanf("%f", &nilaiTugas);
            nilaiAkhir = (0.3 * nilaiUTS) + (0.5 * nilaiUAS) + (0.2 * nilaiTugas);
            printf("\nLAPORAN NILAI MAHASISWA\n");
            printf("NIM : %s\n", nim);
            printf("nama : %s\n", nama);
            printf("nilai akhir : %.2f\n", nilaiAkhir);
            printf("nilai huruf : %c\n", nilaiHuruf(nilaiAkhir));
            return 0; 

        }